core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -228,7 +246,10 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
       // However, in some cases it can pause application threads(STW) for a long moment reading metadata from a physical disk.
       // To prevent this, we forcefully cleanup memory mapping within proper execution which never affects API responsiveness.
       // See https://issues.apache.org/jira/browse/KAFKA-4614 for the details.
-      safeForceUnmap()
+      CoreUtils.swallow(forceUnmap(), this)
+      // Accessing unmapped mmap crashes JVM by SEGV.
+      // Accessing it after this method called sounds like a bug but for safety, assign null and do not allow later access.
+      mmap = null
```

Comment:
forceUnmap sets the `mmap` to `null` so not clear why we need this change at all. I would revert it and test again.